torch_geometric/utils/mixin.py

```diff
@@ -0,0 +1,14 @@
+class CastMixin:
```

Comment:
why do we need this?

Reply:
This is to support the creation of a `TensorAttr` with a tuple or list (as opposed to `TensorAttr(...)`), the use of which can be seen in `test_feature_store.py`.

Reply:
I feel it is a little confusion since in the function we have the type hint indicate what users pass in is a `TensorAttr` but in fact it is `Any` so we need to do the cast. I'd imaging some C++ style like directly cast `{...}` into a class and pass into the function.
I tried to google it and see if there's any best practice but I couldn't find any.

Reply:
Yes, you're absolutely right that the type hints for the functions are technically incorrect as we are able to accept `Any` and cast to the desired class. I'm not sure there is a best practice for this that maintains the same level of code cleanliness.

In particular, I think that the improvements to ease-of-use as a result of `CastMixin` (e.g. indexing via `store[{'tensor_type': x, 'node_type': y}][index]` instead of `store[TensorAttr(tensor_type=x, node_type=y)][index]`) are quite useful. Imo, the current type hints (instead of `Any` everywhere) are provided for improved code readability and clarity, and since they are not enforced it is reasonable to type hint `TensorAttr` and also support non-`TensorAttr` inputs that we subsequently cast. We can add better documentation for `CastMixin` to make this more clear as well. Wdyt?

Reply:
I'm fine with your current design, I think somewhere in the code when we call cast we should keep a comment explain why.